Order methods in generated code by BMI group type

mcflugen opened the pull request:
This pull request makes a couple changes to the *bmipy-render* program. 

* Added a ``--docstring/--no-docstring`` option to optionally leave out the docstrings in the generated code.
* Changed the rendering so that methods of the generated BMI class are grouped by type. I've ordered them so they match what's in the BMI documentation—control functions first, followed by info functions, etc.

mcflugen committed:
sort methods into groups

mcflugen committed:
add --docstring/--no-docstring option to cli

mcflugen committed:
add test for --docstring option

mdpiper reviewed (approved):
Both of these new features (ordered groups and optional docstrings) are awesome! Thank you for making these changes. A couple minor requests:

- The *finalize* method is listed first in `bmipy-render` output. I expected it to be last in the "control" group.
- In `bmipy-render --help`:
  - Can you note that `--docstring` is the default?
  - Add the `help` keyword to the positional parameter?

mcflugen committed:
add default is to include docstrings

mcflugen committed:
order initialize first, then update, then finalize

mcflugen commented:
> The finalize method is listed first in bmipy-render output. I expected it to be last in the "control" group.

Within groups, methods are listed in alphabetical order. I've changed it so that *initialize*, *update*, and *finalize* each have their own groups and so they now appear in the order you would expect.

> Can you note that --docstring is the default?

Yup, I've added that.

> Add the help keyword to the positional parameter?

I'm not sure what you mean. Do you mean changing the usage line so that `--help` appears alongside `-h`? That is, something like,
```
usage: bmipy-render [-h | --help] [--version] [--docstring | --no-docstring] name
```
If so, as odd as it may sound, I'm not sure there is an easy way to do that. Do you know how?

mdpiper commented on src/bmipy/_cmd.py:
```diff
@@ -9,19 +9,29 @@
     """Render a template BMI implementation in Python for class NAME."""
     parser = argparse.ArgumentParser()
     parser.add_argument("--version", action="version", version=f"bmipy {__version__}")
     parser.add_argument("name")
```
I'm sorry, I wasn't clear. Here's what I was thinking:
```
parser.add_argument("name", help="The BMI class name")
```

mcflugen replied:
Ah, thank you. That makes sense. I've made the change.

mcflugen committed:
add a description for the class name argument
